samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp

```diff
@@ -94,17 +94,34 @@ void ExtendedDynamicState2::load_assets()
 {
 	load_scene("scenes/primitives/primitives.gltf");
 
-	std::vector<SceneNode> scene_elements;
-	// Store all scene nodes in a linear vector for easier access
-	for (auto &mesh : scene->get_components<vkb::sg::Mesh>())
+	std::vector<SceneNode>       scene_elements;
+	std::vector<vkb::sg::Node *> node_scene_list = scene->get_root_node().get_children();
```

Comment:
Maybe just initialize the `node_scene_list` with `scene->get_root_node()`?

Reply:
initialization has been changed